library/public/loader.h

```diff
@@ -13,75 +14,25 @@ namespace f3d
  * @brief   Class to load files in F3D
  *
  * A class to load files in F3D.
- * Multiple files and folders can be added using addFiles/addFile and stored
- * in an internal list, with a index pointing to a "current" file.
- * Then files from the list can be loaded one by one using loadFile method.
- * Information about the files in the list can be recovered using getFileInfo.
- * The current file can be changed either on loadFile with a dedicated enum or
- * by using setCurrentFileIndex.
+ * TODO
  */
 class F3D_EXPORT loader
 {
 public:
   /**
-   * An enum used in loadFile to select which file to load.
    */
-  enum class LoadFileEnum
-  {
-    LOAD_FIRST,
-    LOAD_PREVIOUS,
-    LOAD_CURRENT,
-    LOAD_NEXT,
-    LOAD_LAST
-  };
+  virtual bool addGeometry(const std::string& filePath) = 0;
 
   /**
-   * Add a list of files or directories to be loaded.
-   * Set recursive to true to load folder recursively instead of only files.
-   * Duplicate will not be added.
    */
-  virtual loader& addFiles(const std::vector<std::string>& files, bool recursive = false) = 0;
+  virtual loader& resetToDefaultScene() = 0;
 
   /**
-   * Add a file or directory (all files in) to be loaded.
-   * Set recursive to true to load folder recursively instead of only files.
    */
-  virtual loader& addFile(const std::string& path, bool recursive = false) = 0;
+  virtual bool loadFullScene(const std::string& filePath) = 0;
 
-  /**
-   * Get the vector of files to be loaded.
-   */
-  virtual const std::vector<std::string>& getFiles() const = 0;
-
-  /**
-   * Set the current file index, corresponding to the file that will be loaded
-   * when calling loadFile(LOAD_CURRENT).
-   */
-  virtual loader& setCurrentFileIndex(int index) = 0;
-
-  /**
-   * Get the current file index.
-   */
-  virtual int getCurrentFileIndex() const = 0;
-
-  /**
-   * Load a file if any have been added.
-   * Set the load argument to LOAD_FIRST, LOAD_PREVIOUS, LOAD_NEXT or LOAD_LAST to change file
-   * index. This will change the current file index accordingly.
-   * Default is LOAD_CURRENT.
-   * Returns true if a file is loaded successfully, false otherwise.
-   */
-  virtual bool loadFile(LoadFileEnum load = LoadFileEnum::LOAD_CURRENT) = 0;
-
-  /**
-   * Get information about the next file to load according to the load param.
-   * Set the load argument to LOAD_FIRST, LOAD_PREVIOUS, LOAD_NEXT or LOAD_LAST as needed.
-   * nextFileIndex will provide the index of the file or -1 if not available.
-   * filePath the path to the file and fileInfo a more complete information about the file,
-   * including index in list.
-   */
-  virtual void getFileInfo(LoadFileEnum load, int& nextFileIndex, std::string& filePath,
-    std::string& fileName, std::string& fileInfo) const = 0;
+  virtual bool canReadScene(const std::string& filePath) = 0;
+  virtual bool canReadGeometry(const std::string& filePath) = 0;
```

Comment:
see the new API above @snoyer @Meakk

Reply:
Are these `canReadXXX` useful? Just call `addGeometry` and check the result.  
Alternatively you can keep these function but `addGeometry` and `loadFullScene` should return `loader&`

Reply:
> Are these canReadXXX useful? Just call addGeometry and check the result.

What if someone just want to know if a file has a geom reader or a full scene reader without actually reading ?

> Alternatively you can keep these function but addGeometry and loadFullScene should return loader&

How do you know if the file was actually read then ?

Reply:
> What if someone just want to know if a file has a geom reader or a full scene reader without actually reading ?

Then why give the full path? The extension is enough?

> How do you know if the file was actually read then ?

Throw if it fails